### PR TITLE
Maintenance drones no longer get a flash

### LINKED ISSUE #13053: Maintenance Drones have Flashes
**Issue Description**:

Maintenance drones had flashes in their inventory, despite not having them before and that it can only be used to break their laws.

**What did you expect to happen**: 

I think its unintended, as an item that would only break the users laws seems wrong. Drones also didn't have a flash before.

**What happened instead**:

Crack open inventory to find t-ray scanner, find flash

**Why is this bad/What are the consequences**:

See above.

**Steps to reproduce the problem**:

Spawn as drone, check your modules.

**When did the problem start happening**:

Drones didn't have a flash a few days ago, last I played one was maybe two or three days ago.

**Extra information**:

Its probably a drone uprising.

### PR DESCRIPTION
**Pull request by SteelSlayer**
## What Does This PR Do
#12934 Made all `New()` procs call their parent, but in the case of drones, it gave them a flash. This removes their flash, because drones should not have them.

## Why It's Good For The Game
Fixes #13053

## Changelog
:cl:
fix: Maintenance drones no longer get a flash
/:cl:

### CODE REVIEW

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 1]
```diff
@@ -27,9 +27,10 @@
 	..()
 
 
-/obj/item/robot_module/New()
+/obj/item/robot_module/New(loc, should_have_flash = TRUE)
```
Comment (AffectedArc07) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: Sidenote: A lot of these procs missing args will be fixed when I add in dreamchecker, which should kill off a fair few runtimes.

Comment (SteelSlayer) [reply code/modules/mob/living/silicon/robot/robot_modules.dm]: Dreamchecker will be nice

**Comment by Crazylemon64** [comment]
Nah it should be fine that's Initialize that I'm thinking about

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 2]
```diff
@@ -27,9 +27,10 @@
 	..()
 
 
-/obj/item/robot_module/New()
+/obj/item/robot_module/New(loc, should_have_flash = TRUE)
 	..()
-	modules += new /obj/item/flash/cyborg(src)
+	if(should_have_flash)
+		modules += new /obj/item/flash/cyborg(src)
```
Comment (Crazylemon64) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: Wait no what was I thinking
Adding extra items to a module should be put in a separate proc, and then override that proc in the drone not to give the item you don't want them to have

Comment (SteelSlayer) [reply code/modules/mob/living/silicon/robot/robot_modules.dm]: I feel like I should just go ahead and do that for every module, medical, mining, sec etc. Just throw every module's item giving code into some proc, name it `add_module_specific_items` or something, then call that in `New()`

**Comment by Spacemanspark** [comment]
You are droneist sir

Maint drones rise up

**Pull request merged**